core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -84,7 +85,7 @@ object TopicCommand extends Logging {
   private def getTopics(zkClient: KafkaZkClient, opts: TopicCommandOptions): Seq[String] = {
     val allTopics = zkClient.getAllTopicsInCluster.sorted
     if (opts.options.has(opts.topicOpt)) {
-      val topicsSpec = opts.options.valueOf(opts.topicOpt)
+      val topicsSpec = Pattern.quote(opts.options.valueOf(opts.topicOpt))
```

Comment:
I think this will disable the existing regex functionality for describe, alter options.

Reply:
Thanks! You're right. I missed the existing regex capability.
The reported issue exists in describe and alter too. I'll think about this some more and update the PR.

Reply:
@omkreddy Thinking a bit more about this, it seems the existing functionality will be broken with any fix for the reported JIRA. Delete also takes regex strings. The issue is actual topic names and regular expressions are both provided to the same `--topic` option. What do you think?

Reply:
looks like current behavior is correct . "test.topic" treated as regex and topic names like "test1topic" , test-topic",  "test.topic" matches with regex.   

maybe we can just update "--delete" option description to escape special characters , if the user wants provide actual topic name.  
ex: "test\\.topic"

Reply:
Right. IMO, with regex and how they are used we are making it very easy for users to unintentionally delete topics. Ideally, there should be an additional `--regex` option (or something similar) for `--delete` as another level of protection against inadvertent topic deletion with regular expressions.

I agree with what you suggest, even though the same issue also applies to `--describe` and `--alter`. For example, if I have a topic `testttopic`, currently it is removed with both `--topic test.topic` and `--topic test\.topic`. I think only the first one should remove it, and the second one should remove only the `test.topic` (with actual period in the name). I'll try to improve that with this PR.

Reply:
Since we escaping special character, "test\.topic" wont match with "testttopic"

Reply:
Yes, I was talking about the current trunk. Sorry for the confusion.

Reply:
Sorry, I have updated  my previous comment. Current trunk supports  this if we surroung the regex with quotes (```-- topic "test\.topic"```)

Reply:
Right, so if I have two topics `test.topic` and `testttopic` here's how the current trunk works:
1. `--delete --topic test\.topic` or `--delete test.topic` or `--delete "test.topic"` deletes both
2. `--delete --topic "test\.topic"` deletes `test.topic` only

It seems using double quotes and escape characters does not treat the string as a regex. The current implementation favors regex over fixed string. It should ideally be the other way around (doing something like that would require a KIP). Not sure if any changes should be made for this jira though. What do you think?

Reply:
Yes. double quotes allows us to escape special characters, but it is still regex. we can use some thing like this  ```--topic "test\.topic.*"``` to find all topics starting with "test.topic".

As mentioned earlier, maybe we can just update  "--delete" ,  "--topic" option descriptions about regex usage.